tutorials/02_intermediate/gridspec.py

```diff
@@ -1,126 +1,196 @@
+        perhaps the primary function used to create figures and axes.
+        It's also similar to :func:`~matplotlib.pyplot.subplot`,
+        but creates and places all axes on the figure at once.
+
 """
+
 import matplotlib.pyplot as plt
 import matplotlib.gridspec as gridspec
 
-###############################################################################
-# Basic Example of using subplot2grid
-# ===================================
+############################################################################
+# Basic Quickstart Guide
+# ======================
+#
+# These first two examples show how to create a basic 4-by-4 grid using
+# both :func:`~matplotlib.pyplot.subplots` and :mod:`~matplotlib.gridspec`.
+#
+# Using :func:`~matplotlib.pyplot.subplots` is quite simple.
+# It returns a :class:`~matplotlib.figure.Figure` instance and an array of
+# :class:`~matplotlib.axes.Axes` objects.
+
+fig1, f1_axes = plt.subplots(ncols=2, nrows=2)
+fig1.tight_layout()
+
+############################################################################
+# For such a simple use case, :mod:`~matplotlib.gridspec` is perhaps overly
+# verbose.
+# You have to create the figure and :class:`~matplotlib.gridspec.GridSpec`
+# instance separately, then pass the elements gridspec instance to the
+# :func:`~matplotlib.figure.Figure.add_subplot` method to create the axes
+# objects.
+# The elements of the gridspec are accessed as if they where elements of
+# a numpy array of the same shape.
+
+fig2 = plt.figure()
+spec2 = gridspec.GridSpec(ncols=2, nrows=2)
+f2_ax1 = fig2.add_subplot(spec2[0, 0])
+f2_ax2 = fig2.add_subplot(spec2[0, 1])
+f2_ax3 = fig2.add_subplot(spec2[1, 0])
+f2_ax4 = fig2.add_subplot(spec2[1, 1])
+fig2.tight_layout()
+
+#############################################################################
+# When you want to have plots of different sizes, however,
+# :mod:`~matplotlib.gridspec` becomes indispensable and provides a couple
+# of options.
+# The method shown here initializes a uniform grid specification,
+# and then uses typical numpy indexing and slices to allocate multiple
+# "cells" for a given subplot.
+
+fig3 = plt.figure()
+spec3 = gridspec.GridSpec(ncols=3, nrows=3)
+anno_opts = dict(xy=(0.5, 0.5), xycoords='axes fraction',
+                 va='center', ha='center')
+
+fig3.add_subplot(spec3[0, 0]).annotate('GridSpec[0, 0]', **anno_opts)
+fig3.add_subplot(spec3[0, 1:]).annotate('GridSpec[0, 1:]', **anno_opts)
+fig3.add_subplot(spec3[1:, 0]).annotate('GridSpec[1:, 0]', **anno_opts)
+fig3.add_subplot(spec3[1:, 1:]).annotate('GridSpec[1:, 1:]', **anno_opts)
+
+fig3.tight_layout()
+
+############################################################################
+# This example shows off using the ``width_ratios`` and ``height_ratios``
+# options.
+# These keyword arguments are lists of numbers.
+# Note that absolute values are meaningless, only their relative ratios
+# matter.
+# That means that ``width_ratios=[2, 4, 8]`` is equivalent to
```

Comment:
Cool - that makes sense to me! Just noting it to make sure that was intended behavior! (btw, feel free to delete comments of mine that you address or aren't actionable...I tend to do this as a way of ticking off "todo" items in a PR)